### PR TITLE
Trying to simplify the Kafka ingress

### PR DESCRIPTION
**Pull request by worldtiki**

### CODE REVIEW

File: src/main/java/com/hotels/service/tracing/zipkintohaystack/ingresses/kafka/KafkaConsumerConfig.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+import com.hotels.service.tracing.zipkintohaystack.forwarders.Fork;
+import com.hotels.service.tracing.zipkintohaystack.forwarders.haystack.SpanValidator;
+
+@ConditionalOnProperty(name = "pitchfork.ingress.kafka.enabled", havingValue = "true")
+@Configuration
+public class KafkaConsumerConfig {
+
+    @Bean
+    public KafkaConsumer<String, byte[]> kafkaConsumer(KafkaIngressConfig pitchForkConfig) {
```
Comment (nikos912000) [inline src/main/java/com/hotels/service/tracing/zipkintohaystack/ingresses/kafka/KafkaConsumerConfig.java]: Can we change the name of the parameter from `pitchForkConfig` to `kafkaIngressConfig` or even `config` now that we've renamed this file?

File: src/main/java/com/hotels/service/tracing/zipkintohaystack/ingresses/kafka/KafkaConsumerConfig.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        String kafkaBrokers = pitchForkConfig.getBootstrapServers();
+        List<String> sourceTopics = pitchForkConfig.getSourceTopics();
+
+        KafkaConsumer<String, byte[]> consumer = new KafkaConsumer<>(
+                Map.of(
+                        ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, kafkaBrokers,
+                        ConsumerConfig.GROUP_ID_CONFIG, "pitchfork",
+                        ConsumerConfig.AUTO_COMMIT_INTERVAL_MS_CONFIG, "1000",
```
Comment (nikos912000) [inline src/main/java/com/hotels/service/tracing/zipkintohaystack/ingresses/kafka/KafkaConsumerConfig.java]: Could we make these properties configurable (e.g. through `KafkaIngressConfig`)?

File: src/main/java/com/hotels/service/tracing/zipkintohaystack/ingresses/kafka/KafkaRecordsConsumer.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+        Thread thread = new Thread(this::fetchRecordsFromKafka);
+        thread.setDaemon(true);
+        thread.start();
+    }
+
+    private void fetchRecordsFromKafka() {
+        try {
+            while (true) {
+                var records = kafkaConsumer.poll(ofSeconds(1)); // TODO: make poll duration configurable
```
Comment (nikos912000) [inline src/main/java/com/hotels/service/tracing/zipkintohaystack/ingresses/kafka/KafkaRecordsConsumer.java]: Can we raise an issue for that (to make it configurable)? 
Also, I believe it'd be better to use the `ofMillis` in this case.

File: src/main/java/com/hotels/service/tracing/zipkintohaystack/ingresses/kafka/KafkaRecordsConsumer.java [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+                if (!records.isEmpty()) {
+                    StreamSupport.stream(records.spliterator(), false)
+                            .flatMap((Function<ConsumerRecord<String, byte[]>, Stream<Span>>) record -> decoder.decodeList(record.value()).stream())
+                            .filter(spanValidator::isSpanValid)
+                            .forEach(span -> fork.processSpan(span)
+                                    .doOnError(throwable -> logger.warn("operation=fetchRecordsFromKafka", throwable))
+                                    .onErrorResume(e -> Mono.empty())
+                                    .blockLast());
+                    // TODO: consider replacing with a reactor KafkaReceiver
```
Comment (nikos912000) [inline src/main/java/com/hotels/service/tracing/zipkintohaystack/ingresses/kafka/KafkaRecordsConsumer.java]: It'd worth raising an issue for this one.

File: src/main/java/com/hotels/service/tracing/zipkintohaystack/ingresses/kafka/KafkaConsumerConfig.java (hunk 1 above)
Comment (worldtiki) [reply src/main/java/com/hotels/service/tracing/zipkintohaystack/ingresses/kafka/KafkaConsumerConfig.java]: Yeap, makes total sense. Thanks!

File: src/main/java/com/hotels/service/tracing/zipkintohaystack/ingresses/kafka/KafkaConsumerConfig.java (hunk 2 above)
Comment (worldtiki) [reply src/main/java/com/hotels/service/tracing/zipkintohaystack/ingresses/kafka/KafkaConsumerConfig.java]: Yeap!

File: src/main/java/com/hotels/service/tracing/zipkintohaystack/ingresses/kafka/KafkaRecordsConsumer.java (hunk 4 above)
Comment (worldtiki) [reply src/main/java/com/hotels/service/tracing/zipkintohaystack/ingresses/kafka/KafkaRecordsConsumer.java]: I think this is too small for an issue. The current implementation works ok, this is just to flag the existent of tech debt in the 1st one and an improvement on this one.

Comment (nikos912000) [reply src/main/java/com/hotels/service/tracing/zipkintohaystack/ingresses/kafka/KafkaRecordsConsumer.java]: Ok, makes sense!

**Pull request merged**